File: TaxDeductionOrganizer/src/TaxDeductionOrganizer.WebApp/projects/tax-deduction-organizer/src/app/models/tax-year.ts
export interface TaxYear {
  taxYearId: string;
  year: number;
  isFiled: boolean;
  filingDate?: string;
  totalDeductions: number;
  notes?: string;
}

export interface CreateTaxYear {
  year: number;
  notes?: string;
}

export interface UpdateTaxYear {
  taxYearId: string;
  year: number;
  isFiled: boolean;
  filingDate?: string;
  notes?: string;
}

File: TaxDeductionOrganizer/src/TaxDeductionOrganizer.Core/Models/TaxYearAggregate/TaxYear.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace TaxDeductionOrganizer.Core;

public class TaxYear
{
    public Guid TaxYearId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public int Year { get; set; }
    public bool IsFiled { get; set; }
    public DateTime? FilingDate { get; set; }
    public decimal TotalDeductions { get; set; }
    public string? Notes { get; set; }
    public List<Deduction> Deductions { get; set; } = new List<Deduction>();
    
    public void CalculateTotalDeductions()
    {
        TotalDeductions = Deductions.Sum(d => d.Amount);
    }
    
    public void MarkAsFiled()
    {
        IsFiled = true;
        FilingDate = DateTime.UtcNow;
    }
}
